src/commonMain/kotlin/fr/acinq/lightning/wire/LightningMessages.kt

```diff
@@ -1295,3 +1296,22 @@ data class SwapInConfirmed(
         }
     }
 }
+
+@OptIn(ExperimentalUnsignedTypes::class)
+data class PhoenixAndroidLegacyInfo(
+    val hasChannels: Boolean
+) : LightningMessage {
+    override val type: Long get() = PhoenixAndroidLegacyInfo.type
+
+    override fun write(out: Output) {
+       LightningCodecs.writeByte(if (hasChannels) 1 else 0, out)
+    }
```

Comment:
Do you really need to implement the _writing_ part?

Reply:
No we don't really need it and can remove it, although we the test would have to be updated.